module/zfs/sa.c

```diff
@@ -980,7 +980,7 @@ sa_attr_table_setup(objset_t *os, sa_attr_reg_t *reg_attrs, int count)
 	kmem_free(sa->sa_user_table, count * sizeof (sa_attr_type_t));
 	sa->sa_user_table = NULL;
 	sa_free_attr_table(sa);
-	return ((error != 0) ? error : EINVAL);
+	return (error);
```

Comment:
It does look like `error` will be always true.  Given that, in addition to your fix please add an `ASSERT(error != 0);` here before the return.  This way it's clear that `bail` can only be entered on error and it will ensure this behavior doesn't get accidentally changed.

Reply:
Done. That is nice tight code using the ASSERT before return. Is this standard practice within ZFS? I have attempted to mirror the commit format linked to above. I was a little confused as to what was the commit summary and what was the PR title. If it needs tweaking can you please say so, I would like it perfect so the next coverity fixes are done correctly. thanks @behlendorf, it is really cool how quick and responsive ZFS is to new PR's/contributions.

Reply:
We use the ASSERTs in places where there's an invariant that must be true.  If the condition ever fails it will halt the thread and dump a stack trace to the conosle for debugging.  Your system will need to be rebooted after you've gotten whatever additional debugging you need.  They're disabled in production builds and enabled in the debug builds, all the automated testing is done with debug builds.  You'll also see VERIFYs in the code these work the same as ASSERTs but cannot be disabled.

As for the Coverity commit messages we've been using the following form which allows us to batch several small fixes in to a single commit which can be useful.  Here are a few examples, cd5083c00c6215f4f15c04e30feeb5f9457ec896 547c5936613ef296559eda5177b6391cfbbfb5c6 f4bae2ed6361917660f5238cd35672ad18d7babc

```
Fix coverity defects: CID xxx, yyy

CID xxx: Coverity short bug description

Optional description for xxx.  Some bugs are clearly trivial and this can be skipped, but the more subtle bugs should be explained like any other change

CID yyy: Coverity short bug description

Optional description for yyyy

Signed-off-by: ...
```

> really cool how quick and responsive ZFS is to new PR's/contributions.

Thanks, we try!